stream-cipher/src/lib.rs

```diff
@@ -185,3 +210,57 @@ where
         }
     }
 }
+
+/// Trait implemented for numeric types which can be used with the
+/// [`SyncStreamCipherSeek`] trait.
+///
+/// This trait is implemented for the following primitive numeric types:
+/// `u8`, `u16`, `u32`, `i32`, `u64`, and `u128`. It is not intended to
+/// be implemented in third-party crates.
+pub trait SeekNum:
+    Sized
+    + TryInto<u8>
+    + TryFrom<u8>
+    + TryInto<u16>
+    + TryFrom<u16>
+    + TryInto<u32>
+    + TryFrom<u32>
+    + TryInto<i32>
+    + TryFrom<i32>
+    + TryInto<u64>
+    + TryFrom<u64>
+    + TryInto<u128>
+    + TryFrom<u128>
+{
+    /// Try to get position for block number `block`, byte position inside
+    /// block `byte`, and block size `bs`.
+    fn from_block_byte<T: SeekNum>(block: T, byte: u8, bs: u8) -> Result<Self, OverflowError>;
+
+    /// Try to get block number and bytes position for given block size `bs`.
+    fn to_block_byte<T: SeekNum>(self, bs: u8) -> Result<(T, u8), OverflowError>;
+}
```

Comment:
Hmmmm... 😉

My gut reaction to this is there's a lot of complexity here which feels like there's not a valid use case to justify it.

I'm trying to think of cases where it'd be helpful to have something other than a 32-bit or 64-bit number.

In terms of block counts, why would you ever want to use `u8` or `u128`, for example?

I guess it's complete/consistent.

Reply:
I agree that it could be trimmed, but I left it for completeness/symmetricity sake.

>why would you ever want to use u8 or u128

Well, OpenSSL variant of the CTR mode uses 128-bit counters. :)

Reply:
Also it would be nice to use a more compact formatting (i.e. use several bounds per line), but I couldn't find a relevant `rustfmt` option for it.

Reply:
Yeah, the variant of AES-CTR used by AES-CMAC-SIV also uses a 128-bit counter, but that's for determining the initial position.

Data volume limits for most ciphers alone would limit where you can safely seek in the keystream after that.

Reply:
Yeah, but I also seen (private) code which constructs offset based on bits from several sources (i.e. something like 20 bits from `id1`, 30 bits from `id2`, etc.). It could be done by generating nonces instead, but oh well...

I've added signed integers for completeness sake, I think additional code should not noticeably influence compilation times and hopefully one day we will find a way how to make this code more compact.